### PR TITLE
feat: test template from profile examples

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR add ability to prepare test file for specific profile and provider. Result for `prepare:test computer-vision/face-detection google-apis`:

```
import { SuperfaceTest } from '@superfaceai/testing';

describe('computer-vision/face-detection/google-apis', () => {
  let superface: SuperfaceTest;

  beforeEach(() => {
    superface = new SuperfaceTest({
      profile: 'computer-vision/face-detection',
      provider: 'google-apis',
    });
  });

  describe('FaceDetection', () => {
    // specify test case name
    it('should map error', async () => {
      const result = await superface.run({
        useCase: 'FaceDetection',
        input: {
          imageUrl : 'https://upload.wikimedia.org',
        }
      })
      
      expect(result.isErr()).toBe(true);
      expect(result).toMatchSnapshot();
    });
  
    // specify test case name
    it('should perform successfully', async () => {
      const result = await superface.run({
        useCase: 'FaceDetection',
        input: {
          imageUrl : 'https://upload.wikimedia.org/wikipedia/commons/c/c3/RH_Louise_Lillian_Gish.jpg',
        }
      })
      
      expect(result.isOk()).toBe(true);
      expect(result).toMatchSnapshot();
    });
  });  
});
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [ ] I have read the **CONTRIBUTION_GUIDE** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by tmladek — CHANGES_REQUESTED** [review]
```
    it('should map error', async () => {
      const result = await superface.run({
        useCase: 'FaceDetection',
        input: {
          code : '14',
          message : 'We can not access the URL currently. Please download the content and pass it in.',
        }
      })
```
      
I don't think this is correct; `input` [should correspond to](https://github.com/superfaceai/station/blob/main/grid/computer-vision/face-detection/profile.supr#L166) `imageUrl = 'https://upload.wikimedia.org',`

**Comment by martinalbert** [comment]
I would also consider specifying test instance in constructor since it's already generated for jest.

File: src/templates/prepared-test/test-templates/error.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export default `// specify test case name
       {{/unless}}
     })
     
-    expect(result.isErr()).toBe(true);
+    expect(() => result.unwrap()).not.toThrow();
```
Comment (tmladek) [inline src/templates/prepared-test/test-templates/error.ts:13]:
```suggestion
    expect(() => result.unwrap()).toThrow();
```

**Review by tmladek — CHANGES_REQUESTED** [review]

**Review by kysely — APPROVED** [review]
I like this, thanks 🙏 

A question though: I see it's becoming a pattern [in Station to define provider-agnostic test](https://github.com/superfaceai/station/blob/main/grid/language/analyze-plain-text-sentiment/maps/analyze-plain-text-sentiment.ts) which is then ran against all providers.

Is this the way we'd like to go forward with in the future? @tmladek

**Pull request merged**